cellengine/resources/gate.py

```diff
@@ -162,81 +144,347 @@ def create(
+            y2 (List[float], optional): The second y coordinate (after the
+                    channel's scale has been applied).
+            label (float, optional): Position of the label. Defaults to the
+                midpoint of the gate. gid (str): Group ID of the gate, used for
+                tailoring. If this is not specified, then a new Group ID will
+                be created. To create a tailored gate, the gid of the global
+                tailored gate must be specified.
+            locked (bool, optional): Prevents modification of the gate via the
+                web interface.
+            tailored_per_file (bool): Whether or not this gate is tailored per
+                FCS file.
+            fcs_file_id (str, optional): ID of FCS file, if tailored per file.
+                Use ``None`` for the global gate in a tailored gate group. If
+                specified, do not specify ``fcs_file``.
+            fcs_file (str, optional): Name of FCS file, if tailored per file.
+                An attempt will be made to find the file by name. If zero or
+                more than one file exists with the name, an error will be
+                thrown. Looking up files by name is slower than using the ID,
+                as this requires additional requests to the server. If
+                specified, do not specify ``fcs_file_id``.
 
-class PolygonGate(Gate):
-    """Basic concrete class for polygon gates"""
+        Returns:
+            A RectangleGate object.
+
+        Example:
+            ```python
+            experiment.create_rectangle_gate(x_channel="FSC-A", y_channel="FSC-W",
+            name="my gate", 12.502, 95.102, 1020, 32021.2)
+            cellengine.Gate.create_rectangle_gate(experiment_id, x_channel="FSC-A",
+            y_channel="FSC-W", name="my gate", x1=12.502, x2=95.102, y1=1020,
+            y2=32021.2, gid=global_gate.gid)
+            ```
+        """
+        kwargs = get_args_as_kwargs(cls, locals())
+        params = {
+            k: kwargs.pop(k) for k in ["create_population", "parent_population_id"]
+        }
+
+        gate = cls._format(**kwargs)
+        return ce.APIClient().create(Gate(id=None, **gate), **params)  # type: ignore
+
+    @classmethod
+    @exception_handler
+    def _format(cls, **kwargs):
+        """Get relevent kwargs and shape into a gate model"""
+
+        kwargs = remove_keys_with_none_values(kwargs)
+
+        x1 = kwargs.get("x1") or kwargs.get("model", {}).get("rectangle", {}).get("x1")
+        x2 = kwargs.get("x2") or kwargs.get("model", {}).get("rectangle", {}).get("x2")
+        y1 = kwargs.get("y1") or kwargs.get("model", {}).get("rectangle", {}).get("y1")
+        y2 = kwargs.get("y2") or kwargs.get("model", {}).get("rectangle", {}).get("y2")
+        label = kwargs.get("label") or kwargs.get("model", {}).get(
+            "label",
+            [
+                mean([x1, x2]),  # type: ignore
+                mean([y1, y2]),  # type: ignore
```

Comment:
I'll probably ask on most of these. Why is `# type: ignore` needed here?

Reply:
Cleaned up some other typing errors however